plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -195,4 +220,24 @@ public static Object[][] timestampTypes()
                 {"timestamp(12)"}
         };
     }
+
+    @DataProvider
+    public Object[][] testTableNameDataProvider()
+    {
+        return testTableNameTestData().stream()
+                .collect(toDataProvider());
+    }
+
+    private List<String> testTableNameTestData()
+    {
+        return ImmutableList.<String>builder()
+                .add("tricky")
```

Comment:
What about renaming and reusing `io.trino.testing.BaseConnectorTest#testColumnNameTestData`?
What about having this as a generic test (in BCT) to guard all connectors against "tricky table names"?

We probably would want separate method to `filterColumnNameTestData` method, since column naming rules are not necessarily same as table naming rules in the remote system. For example, some databases store tables in files with names after table names, so file system naming constraints apply.

cc @hashhar @ebyhr @kokosing

Reply:
> What about renaming and reusing `io.trino.testing.BaseConnectorTest#testColumnNameTestData`?

I think these rules a little bit different for table name and column name, so data provider should be different for such cases and tests should be different I plan to use this test in this PR:
https://github.com/trinodb/trino/pull/14272

By overriding ` testColumnNameTestData`

Reply:
> What about having this as a generic test (in BCT) to guard all connectors against "tricky table names"?

Maybe we can move this test to BCT and then just override Data Provider for SQLServer

Reply:
We should test tricky names for all connectors - would help make sure of catching other "special" databases as well.

I'd propose keeping test + data provider both in BCT and see if something else fails.

As for column and table name rules being different we can have single data provider + two different filter methods so that other connectors can either filter for column names or table names depending on what is not supported.

Reply:
> I think these rules a little bit different for table name and column name, so data provider should be different for such cases and tests should be different

yes, but source of "tricky names" can be the same

Reply:
agree. I mean single data provider + two filter methods (one for table and one for columns) - both do not filter anything in default impl.

Reply:
absolutely, no unified filtering (https://github.com/trinodb/trino/pull/14286#discussion_r984360633)

Reply:
Okay to doing this in a followup.
let's add a TODO comment on this method.